src/dbwrapper.cpp

```diff
@@ -97,46 +97,59 @@ static void SetMaxOpenFiles(leveldb::Options *options) {
              options->max_open_files, default_open_files);
 }
 
-static leveldb::Options GetOptions(size_t nCacheSize)
+static void GetOptions(size_t nCacheSize, leveldb::Options* options)
 {
-    leveldb::Options options;
-    options.block_cache = leveldb::NewLRUCache(nCacheSize / 2);
-    options.write_buffer_size = nCacheSize / 4; // up to two write buffers may be held in memory simultaneously
-    options.filter_policy = leveldb::NewBloomFilterPolicy(10);
-    options.compression = leveldb::kNoCompression;
-    options.info_log = new CBitcoinLevelDBLogger();
+    options->block_cache = leveldb::NewLRUCache(nCacheSize / 2);
+    options->write_buffer_size = nCacheSize / 4; // up to two write buffers may be held in memory simultaneously
+    options->filter_policy = leveldb::NewBloomFilterPolicy(10);
+    options->compression = leveldb::kNoCompression;
+    options->info_log = new CBitcoinLevelDBLogger();
     if (leveldb::kMajorVersion > 1 || (leveldb::kMajorVersion == 1 && leveldb::kMinorVersion >= 16)) {
         // LevelDB versions before 1.16 consider short writes to be corruption. Only trigger error
         // on corruption in later versions.
-        options.paranoid_checks = true;
+        options->paranoid_checks = true;
     }
-    SetMaxOpenFiles(&options);
-    return options;
+    SetMaxOpenFiles(options);
+}
```

Comment:
I agree with @ryanofsky. I don't think there's a strong need to change this function, but if there is, I would prefer it to return the target type `std::unique_ptr<leveldb::Options, void (*)(leveldb::Options*)` directly, instead of using a pointer out argument (which, strictly, seems a deterioration of API clarity to me).

Reply:
Will work on a patch with this, thanks for the review

Reply:
Latest change lifts the `GetOptions` logic into the ctor of `LevelDbOptions`